src/darwin/Framework/CHIP/MTRCommissioningOperation.mm

```diff
@@ -250,8 +251,22 @@ - (void)controller:(MTRDeviceController *)controller statusUpdate:(MTRCommission
     });
 }
 
-- (void)controller:(MTRDeviceController *)controller commissioningSessionEstablishmentDone:(NSError * _Nullable)error
+- (void)controller:(MTRDeviceController *)controller commissioningSessionEstablishmentDone:(NSError * _Nullable)error forParameters:(const std::optional<chip::RendezvousParameters> &)parameters
 {
+    if (!error && parameters) {
+        std::vector<SetupPayload> payloads;
+        CHIP_ERROR err = SetupPayload::FromStringRepresentation(_setupPayload.UTF8String, payloads);
+        if (err == CHIP_NO_ERROR) {
+            // Look for a payload matching the RendezvousParameters.
+            for (auto & payload : payloads) {
```

Comment:
Isn't this same sort of logic already in the SetupCodePairer? Or is it just similar...

Reply:
In the pairer we look for an existing payload that matches the discriminator.

I did consider having the pairer communicate the payload it used to us directly, but the problem is that the delegate API used here is technically called by the device controller, not the pairer.  But now that I think about it, maybe that's OK: the _pairer_ (as opposed to the autocommissioner) is always used by the controller, so we could have an API that's called only in the cases when the "here is a string" commissioning APIs are used.  I'll go ahead and add that to core, then implement on top of it here, which will make this simpler.